File: 1653.minimum-deletions-to-make-string-balanced/solution.go
func minimumDeletions(s string) int {
    a := 0
    b := 0
    res := len(s)

    for _, ch := range s {
        if ch == 'a' {
            a++
        }
    }
    for _, ch := range s {
        if ch == 'a' {
            a--
        } 
        res = min(res, a + b)
        if ch == 'b' {
            b++
        }
    }
    return res
}

File: 1653.minimum-deletions-to-make-string-balanced/solution.kt
class Solution {
    fun minimumDeletions(s: String): Int {
        var res = s.length
        var a = s.count { ch -> ch == 'a' }
        var b = 0
        s.forEach { ch ->
            if (ch == 'a') {
                a--
            }
            res = minOf(res, a + b)
            if (ch == 'b') {
                b++
            }
        }
        return res
    }
}